### PR TITLE
add gh pages

### LINKED ISSUE #227: Update github workflow to include staging and develop
## Description
We need a way to easily preview changes ideally when they are in PR mode. The current approach of using GitHub Pages for testing changes in a non-live environment is not ideal because it requires merging all changes into the staging branch. This becomes problematic when multiple changes need to be tested independently, as they cannot be isolated for separate testing.

**Note**
Cloudflare is seen as a viable solution that to its usage in other projects. So this investigation will focus on leveraging cloudflare to handle deployment of the preview links.

#### Tasks
- [x] Investigate a workable deployment process for each PR so that a preview URL can be generated. and **commented** into the PR.
- [x] Investigate the use of Cloudflare's Wrangler based deployments. 
- [x] Investigate the use of Github actions with Cloudflare as the _deployment_ step

## Deployment using Cloudflare
Cloudflare does not have Quarto as a preset framework. So a config file like Wrangler is required. Wrangler is similar in nature to Github actions but provides less flexibility. We can define a set of variable states and "tell" cloudflare what to run. 
There were some problems with Wrangler. There are 3 different versions but more importantly Cloudflare Pages cannot use any of the settings specified in the wrangler documentation other than what is stated on [this](https://developers.cloudflare.com/pages/configuration/build-configuration/) page.

My investigation showed that no scripts can be run directly inside wrangler due to permissions issues. Therefore Quarto could not be installed. This means that leveraging Wrangler to install Quarto and render the markdown _on_ Cloudflare is not possible. Cloudflare needs to be provided with the compiled static content _after_ Quarto has rendered it

The Investigation code commit is available [here](https://github.com/autonity/docs.autonity.org/commit/1f34f7e77310f2b57e099c29508df46b44d9d603)

``` sh
15:12:39.026	sudo: The "no new privileges" flag is set, which prevents sudo from running as root.
15:12:39.026	sudo: If sudo is running in a container, you may need to adjust the container configuration to disable the flag.
15:12:39.029	Failed: Error while executing user command. Exited with error code: 1
```

``` toml
name = "docs-autonity-org"
pages_build_output_dir = "./docs"
build = "wget https://github.com/quarto-dev/quarto-cli/releases/download/v1.5.56/quarto-1.5.56-linux-amd64.deb && sudo dpkg -i quarto-${QUARTO_VERSION}-linux-amd64.deb && quarto render"
```

## Deployment using Github Actions

The problem with the previous approach was that Cloudflare cannot compile in the cloud and therefore needs rendered content. 

This approach leverages Github actions to handle the rendering of Quarto via Quarto's actions [1](https://github.com/quarto-dev/quarto-actions/tree/v2/setup), [2](https://github.com/quarto-dev/quarto-actions/tree/v2/render), with Cloudflare's action[3](https://github.com/cloudflare/pages-action/tree/v1/) used to perform the deployment and provide the output. And finally add-pr-comment[4](https://github.com/mshick/add-pr-comment/tree/v2/) is used to create the comment, leveraging the generated output from Cloudflare's action.

#### Problems with this setup.

Following testing on PR #231. Investigation showed that Cloudflare would not "comment" directly on the PR and provide a preview link, so an additional step is required to take the generated [output]([5](https://github.com/cloudflare/pages-action/tree/v1/?tab=readme-ov-file#outputs) ) from the cloudflare step and use it to post on the PR.

PR #231 has an [action](https://github.com/autonity/docs.autonity.org/blob/cloudpages-integration/.github/workflows/cloudflare-preview.yml) that is currently implementing this approach.

#### Recommended approach

I suggest we test the GitHub Actions approach

1.  Leverage Quarto's actions to prepare and render the site
2. Use Cloudflares actions to create the preview and generate the output necessary to provide the url.
3. Use a pr-comment action to comment out the url to the PR.

As part of the investigation process, PR #231 has been created using github actions that contain these steps.

---
#### Legacy Information (for Reference Only):
---
###   Description

With the addition of new features to the documentation site, it would be beneficial to set up a staging server. This will allow us to thoroughly test new changes before pushing them live. 

Additionally, it makes sense to establish a develop branch for feature development. Changes can be pushed to develop for testing before merging into master, aligning our workflow with best practices used in our other projects. 

- [x] Add develop and staging branches (off of master)
- [x] Configure repo for gh_pages
- [x] update readme development workflow

### PR DESCRIPTION
**Pull request by dweng0**
This PR is part of <https://github.com/autonity/docs.autonity.org/issues/227>. It updates the gh_pages workflow so that any merges to the staging server will be published to gh_pages for testing and review.

- **update the gh_pages workflow**
- **update readme** (also includes link to ghpage)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,11 +45,13 @@ The repo is configured to publish:
...
+- The `master` branch will always contain the live production version of the website.
+- The `develop` branch will be used for ongoing development and feature testing. All changes should be pushed to `develop` first, where they can be tested on a staging server before being merged into `master`.
+- Hot-fixes to the production site should be handled by creating a branch from `master`, implementing the fix, and creating a PR. Once the hot-fix is merged, `master` should be merged into `develop` to ensure the fix is propagated and conflicts are resolved.
+- A new version of the production website is published by merging `develop` into `master` after thorough testing.
 
+The staging site is available [here](https://autonity.github.io/docs.autonity.org/). The staging site is updated automatically on every push to the `staging` branch.
+  
 See the repo Wiki Page [Git Workflow](https://github.com/autonity/game.autonity.org/wiki/Git-Workflow) for [when to `merge` or `rebase` branches when working on `game.autonity.org`](https://github.com/autonity/game.autonity.org/wiki/Git-Workflow#when-to-merge-and-rebase-when-working-on-gameautonityorg) repo.
```
Comment (jay-cousins) [inline README.md:55]: The link to game.autonity.org doesn't resolve, so we need to eitherremove this or copy in the content directly?

**Review by jay-cousins — CHANGES_REQUESTED** [review]
Just one comment on the git workflow

File: README.md (hunk 1 above)
Comment (dweng0) [reply README.md:55]: Agreed. But i think as this is code that hasn't been modified in this PR it should be done in a seperate PR

**Review by jay-cousins — APPROVED** [review]
LGTM

**Pull request merged**